Detectors/TRD/simulation/include/TRDSimulation/Digitizer.h

```diff
@@ -55,11 +55,11 @@ class Digitizer
 
   void getHitContainerPerDetector(const std::vector<HitType>&, std::array<std::vector<HitType>, kNdet>&);
   // Digitization chaing methods
-  bool convertHits(const int, const std::vector<HitType>&, SignalContainer_t&); // True if hit-to-signal conversion is successful
-  bool convertSignalsToDigits(const int, SignalContainer_t&);                   // True if signal-to-digit conversion is successful
-  bool convertSignalsToSDigits(const int, SignalContainer_t&);                  // True if signal-to-sdigit conversion is successful
-  bool convertSignalsToADC(const int, SignalContainer_t&);                      // True if signal-to-ADC conversion is successful
-  bool diffusion(float, double, double, double&, double&, double&);             // True if diffusion is applied successfully
+  bool convertHits(const int, const std::vector<HitType>&, std::vector<ADC_t>&); // True if hit-to-signal conversion is successful
```

Comment:
This is really a completely flat vector which seems not the most convenient .... However, since `SignalContainer_t == map<key, std::array<>>`, I had more in mind that we can model the signal output as `std::vector<std::pair<key, std::array<> >`. In this case, almost nothing would change in the digit creation. We can still loop over the elements, fetch the key and the signal without any costly map operations. The harder part will be do create this structure, which might need an additional lookup of the vector index where in the vector a certain key is located (without search). This can be used either internally in convertHits or can be passed around. I would be ok proposing a PR that implements these ideas and then we can discuss.